src/mesh/BoundingBox.cpp

```diff
@@ -13,50 +13,63 @@ namespace precice::mesh {
 
 logging::Logger BoundingBox::_log{"mesh::BoundingBox"};
 
+BoundingBox::BoundingBox(Eigen::VectorXd boundMin, Eigen::VectorXd boundMax)
+{
+  // PRECICE_ASSERT 1
+  PRECICE_ASSERT(boundMin.rows() == boundMax.rows(), "Dimension of min ({}) and max ({}) vertices should be the same.", boundMin.rows(), boundMax.rows());
+
+  // PRECICE_ASSERT 2
+  PRECICE_ASSERT((boundMin - boundMax).maxCoeff() < 0, "Each component of min vertex ({}) must be <= max vertex ({}) in the same axis direction.", boundMin, boundMax);
+
+  _boundMin   = std::move(boundMin);
+  _boundMax   = std::move(boundMax);
```

Comment:
Is this a performance comment? Pass-by.reference doesn't make sense here, as we anyway want to store the data in the class. If we pass an rvalue into the constructor, we should be able to directly move it into the class. If we pass an lvalue, we make a copy. If we pass-by-reference, we would make a copy in any case.

Reply:
> Pass-by.reference doesn't make sense here, as we anyway want to store the data in the class.

Can't we do something like this, to avoid copying? I am not sure if it would even compile.

```c++
BoundingBox::BoundingBox(Eigen::VectorXd& boundMin, Eigen::VectorXd& boundMax)
:
_boundMin(std::move(boundMin)), _boundMax(std::move(boundMax))
{
// ...
}
```

But anyway, performance-wise it does not make any sense (this is happening only once and the vectors are trivially small). My main point was that I did not really like stealing the resource in the middle of the body. I would prefer if such an operation was in the initializer list (and generally trying to stick to prefer initializers, as e.g., https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-initialize suggests)

You can safely ignore my comment.

Reply:
You could, but the user (we) would be very surprised if the vectors are all of a sudden empty when you construct a bounding box out of them.

Reply:
But of course we can first put the arguments in the initializer list and still keep the PRECICE_ASSERTs afterwards.

Reply:
Isn't the same happening already? That is also bad, but I assumed it was ok so far.

Reply:
> Isn't the same happening already? That is also bad, but I assumed it was ok so far.

No, there are no references involved in the constructor right now, i.e., we first copy the data into the local scope and afterwards move it to the class members. If you pass arguments as an rvalue, the compiler is (usually) able to get rid of the unnecessary copy (but only when it is an rvalue).

Reply:
Ah, right. Just disregard my comment, the current approach is better.